### PR TITLE
Fix case sensitive trashbin_skip_extensions

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Make trashbin_skip_extensions setting case insensitive

## Related Issue

- Fixes https://github.com/owncloud/core/issues/38952

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiShareCreateSpecialToShares1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/64/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavLocks2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/94/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiCapabilities-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/55/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiAuthOcs-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/53/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiWebdavMove1-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/97/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiVersions-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/90/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProvisioning-v2-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/60/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingNotify-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/157/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-6-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/176/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingPublic2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/151/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiProxySmoke-8-8-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/178/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt1-latest-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/160/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIProxySmoke-3-2-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31015/169/1

File: tests/acceptance/features/apiTrashbin/trashbinSkip.feature [hunk 1]
```diff
@@ -39,6 +40,71 @@ Feature: files and folders can be deleted completely skipping the trashbin
...
+    And using <dav-path> DAV path
+    When user "Alice" deletes the following files
+      | path       |
+      | sample.TXT |
+      | sample.DAT |
+      | sample.PHP |
+      | sample.GO  |
+      | sample.PY  |
+    Then as "Alice" the file with original path "/sample.TXT" should exist in the trashbin
+    Then as "Alice" the file with original path "/sample.txt" should exist in the trashbin
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinSkip.feature:69]:
```suggestion
    And as "Alice" the file with original path "/sample.txt" should exist in the trashbin
```

File: tests/acceptance/features/apiTrashbin/trashbinSkip.feature [hunk 2]
```diff
@@ -39,6 +40,71 @@ Feature: files and folders can be deleted completely skipping the trashbin
...
+    And as "Alice" the file with original path "/sample.DAT" should exist in the trashbin
+    And as "Alice" the file with original path "/sample.dat" should exist in the trashbin
+    And as "Alice" the file with original path "/sample.PHP" should exist in the trashbin
+    And as "Alice" the file with original path "/sample.php" should exist in the trashbin
+    And as "Alice" the file with original path "/sample.GO" should exist in the trashbin
+    And as "Alice" the file with original path "/sample.go" should exist in the trashbin
+    And as "Alice" the file with original path "/sample.PY" should exist in the trashbin
+    And as "Alice" the file with original path "/sample.py" should exist in the trashbin
```
Comment (phil-davis) [inline tests/acceptance/features/apiTrashbin/trashbinSkip.feature:77]:
```suggestion
    And as "Alice" the file with original path "/sample.PY" should exist in the trashbin
    And as "Alice" the file with original path "/sample.py" should exist in the trashbin
    But as "Alice" the file with original path "/sample.DAT" should not exist in the trashbin
    And as "Alice" the file with original path "/sample.dat" should not exist in the trashbin
    And as "Alice" the file with original path "/sample.PHP" should not exist in the trashbin
    And as "Alice" the file with original path "/sample.php" should not exist in the trashbin
    And as "Alice" the file with original path "/sample.GO" should not exist in the trashbin
    And as "Alice" the file with original path "/sample.go" should not exist in the trashbin
```

**Review by phil-davis — COMMENTED** [review]
The test scenario will need to be adjusted to expect the new case-insensitive behavior.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt2-10.6.0-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31016/164/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt2-latest-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31016/163/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiTrashbin-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/31020/88/1

**Comment by phil-davis** [comment]
@JanAckermann I will push another fix for the test scenario...

**Comment by phil-davis** [comment]
IMO it will pass now - the scenario passes locally for me.
I had to add another `strtolower` so that both things are lower-cased before comparing.

@JanAckermann maybe you want to squash the commits.

File: tests/acceptance/features/apiTrashbin/trashbinSkip.feature [hunk 3]
```diff
@@ -121,6 +216,21 @@ Feature: files and folders can be deleted completely skipping the trashbin
       | old      |
       | new      |
 
+  @issue-38952 @skipOnOcV10
```
Comment (JanAckermann) [inline tests/acceptance/features/apiTrashbin/trashbinSkip.feature:219]: @phil-davis should remove this based on comment: https://github.com/owncloud/core/issues/38952#issuecomment-876323508

Comment (phil-davis) [reply tests/acceptance/features/apiTrashbin/trashbinSkip.feature:219]: According to that comment, this scenario is expecting the "wrong" behavior. We need to make it like the scenarios in `tests/acceptance/features/apiTrashbin/trashbinSkipOc10Isssue38952.feature` that demonstrate the current behavior. (And delete that other `trashbinSkipOc10Isssue38952.feature`)

Same for the other scenario that demonstrates behavior for file extensions inside folders.

Do you want me to sort it out?

**Review by phil-davis — APPROVED** [review]

**Pull request merged**